bin/deploy-vhost

```diff
@@ -133,112 +118,55 @@ sub get_pgpw_string {
-    }
-    my $db_host = get_db_host($params, $internal);
-    my $pgpw_string = get_pgpw_string($params, $database);
-
-    my $conf = <<END;
-define('OPTION_$params->{prefix}_DB_HOST', '$db_host');
-$port_line
-define('OPTION_$params->{prefix}_DB_NAME', '$database');
-define('OPTION_$params->{prefix}_DB_USER', '$database');
-define('OPTION_$params->{prefix}_DB_PASS', '\${\\(pgpw('$pgpw_string'))}');
-END
-    return $conf;
-
-}
-
-sub one_db_conf_yml {
-    my ($params, $database, $internal) = @_;
-    my $port_line = '';
-    if ($params->{type} eq 'psql') {
-        $port_line = "$params->{prefix}_DB_PORT: $params->{port}";
-    }
-    my $db_host = get_db_host($params, $internal);
-    my $pgpw_string = get_pgpw_string($params, $database);
-
-    my $conf = <<END;
-$params->{prefix}_DB_HOST: '$db_host'
-$port_line
-$params->{prefix}_DB_NAME: '$database'
-$params->{prefix}_DB_USER: '$database'
-$params->{prefix}_DB_PASS: '\${\\(pgpw('$pgpw_string'))}'
-END
-    return $conf;
-
-}
 
+# For each database, pull in information from the JSON and derive some
+# further data as required. Store this in $conf->{database_configs}
 sub db_conf {
     my ($params, $database) = @_;
     if ($params->{type} ne 'mysql' && $params->{type} ne 'psql' && $params->{type} ne 'mongo'){
         die "unknown database type '$params->{type}'";
     }
-
-    # Strings for PHP configuration
-    $conf->{database_configs}{default} .= one_db_conf_php($params, $database, 1);
-    $conf->{database_configs}{external} .= one_db_conf_php($params, $database, 0);
-
-    # Strings for YAML configuration
-    $conf->{database_configs}{yaml} .= one_db_conf_yml($params, $database, 1);
-    $conf->{database_configs}{external_yaml} .= one_db_conf_yml($params, $database, 0);
-
     my $pgpw_string = get_pgpw_string($params, $database);
 
-    # Strings for Rails configuration
-    if ($params->{type} eq 'psql') {
-            # XXX doesn't cope with multiple Rails database configs (as you
-            # have to list them in the yml file in right place, so one text
-            # option can't), oh well
-            if (scalar(@{$conf->{'databases'}}) == 1) {
-                my $db_host = get_db_host($params, $internal_network);
-                $conf->{database_configs}{rails} = <<END;
-    adapter: postgresql
-    database: $database
-    username: $database
-    password: \${\\(pgpw('$pgpw_string'))}
-    host: $db_host
-    port: $params->{port}
-END
+    my $db_config = {
+        prefix   => $params->{prefix},
+        host     => get_db_host($params),
+        name     => $database,
+        username => $database,
+        password => "\${\\(pgpw('$pgpw_string'))}",
+    };
+
+    if ($params->{port}) {
+        $db_config->{port} = $params->{port};
+        if ($params->{replica}) {
+            $db_config->{replica_host} = "$params->{replica}.ukcod.org.uk";
+            if ($params->{port} eq '5432') {
+                $db_config->{replica_port} = 5433;
```

Comment:
Feels like this list of port mappings should be a hash, or a function in Deploy.pm that can be called here.